### PR TITLE
Update Azure SDK Libraries in Kanister

### PR DESCRIPTION
**Pull request by mabhi**
## Change Overview

As per Azure guidelines, few modules from azure library are retiring support from 31 March 2023 . Due to which there is need to replace such modules with azure recommended azure-go-sdk modules.

`go-autorest` being used in the `blockstorage` package will be out of support from March 2023. Follow the documentation below to replace it with a new supported package. 
https://azure.microsoft.com/en-us/updates/support-for-azure-sdk-libraries-that-do-not-conform-to-our-current-azure-sdk-guidelines-will-be-retired-as-of-31-march-2023/

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [x] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [x] :robot: Test

## Issues 

- fixes [#1568](https://github.com/kanisterio/kanister/issues/1568)

## Test Plan

- [ ] :muscle: Manual
- [x] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -199,3 +199,11 @@ require (
...
 	sigs.k8s.io/structured-merge-diff/v4 v4.2.3 // indirect
 )
+
+require (
+	github.com/Azure/azure-sdk-for-go/sdk/azidentity v1.3.1 // indirect
+	github.com/AzureAD/microsoft-authentication-library-for-go v1.2.0 // indirect
+	github.com/golang-jwt/jwt/v5 v5.0.0 // indirect
+	github.com/kylelemons/godebug v1.1.0 // indirect
+	github.com/pkg/browser v0.0.0-20210911075715-681adbf594b8 // indirect
+)
```
Comment (pavannd1) [inline go.mod:209]: We should move this into the group of indirect modules above

File: pkg/blockstorage/azure/azuredisk.go [hunk 2]
```diff
@@ -13,9 +13,11 @@ import (
 
 	"github.com/Azure/azure-sdk-for-go/profiles/latest/compute/mgmt/skus"
 	"github.com/Azure/azure-sdk-for-go/profiles/latest/resources/mgmt/subscriptions"
+	azto "github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
 	azcompute "github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2019-03-01/compute"
 	"github.com/Azure/azure-sdk-for-go/storage"
-	azto "github.com/Azure/go-autorest/autorest/to"
+	helper "github.com/kanisterio/kanister/pkg/blockstorage"
```
Comment (pavannd1) [inline pkg/blockstorage/azure/azuredisk.go:19]:
```suggestion
```
This is not required. It's already imported on line 24. You can replace `helper.*` to `blockstorage.*`

File: pkg/blockstorage/azure/client.go [hunk 3]
```diff
@@ -20,11 +20,13 @@ package azure
 
 import (
 	"context"
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/azure/auth"
```
Comment (pavannd1) [inline pkg/blockstorage/azure/client.go:24]: Looks like VSCode added these back. The client code could still be using these libraries.

**Review by pavannd1 — COMMENTED** [review]
I know this is WIP. Providing some early feedback

File: pkg/blockstorage/azure/azuredisk.go (hunk 2 above)
Comment (mabhi) [reply pkg/blockstorage/azure/azuredisk.go]: Addressed.

File: go.mod (hunk 1 above)
Comment (mabhi) [reply go.mod]: Addressed

**Comment by mabhi** [comment]
moving to https://github.com/kanisterio/kanister/pull/2368

**Pull request closed without merging**